### PR TITLE
Jetpack:  Update disconnect jetpack to redux

### PR DESCRIPTION
**Pull request by enejb**
Currently the disconnection flow in calypso doesn't use Redux. 
This PR adds the necessary actions and reduces to disconnect Jetpack sites.

### CODE REVIEW

**Comment by ebinnion** [comment]
cc @dmsnell since he mentioned a new data layer that we may want to use that results with less code and easier tests.

**Comment by ebinnion** [comment]
This looks reasonable to me. Tests pass. It doesn't look like it's being used anywhere, so let's go ahead and ship. 🚢

File: client/state/jetpack/connection/reducer.js [hunk 1]
```diff
@@ -47,7 +50,14 @@ export const requests = createReducer( {}, {
 	[ JETPACK_CONNECTION_STATUS_REQUEST_SUCCESS ]: createRequestReducer( false )
 } );
 
+export const disconnectsRequests = createReducer( {}, {
```
Comment (tyxla) [inline client/state/jetpack/connection/reducer.js]: I'd probably name this `disconnectRequests`, but that's definitely not a blocker.

File: client/state/jetpack/connection/test/actions.js [hunk 2]
```diff
@@ -89,4 +94,73 @@ describe( 'actions', () => {
...
+			useNock( ( nock ) => {
+				nock( 'https://public-api.wordpress.com:443' )
+					.persist()
+					.post( '/rest/v1.1/jetpack-blogs/' + siteId + '/mine/delete' )
+					.reply( 200, status, {
+						'Content-Type': 'application/json'
+					} );
+			} );
+
+			it( 'should dispatch a request connection status action when thunk triggered', () => {
```
Comment (tyxla) [inline client/state/jetpack/connection/test/actions.js]: It probably makes sense to reword this as `disconnect request action` instead of `request connection status`.

File: client/state/jetpack/connection/test/reducer.js [hunk 3]
```diff
@@ -146,4 +150,72 @@ describe( 'reducer', () => {
...
 		} );
 	} );
+
+	describe( 'disconnectsRequests', () => {
+		it( 'state should default to an empty object', () => {
+			const state = disconnectRequestsReducer( undefined, {} );
+			expect( state ).to.eql( {} );
+		} );
+
+		it( 'should set requesting to true for the specified site when status request starts', () => {
```
Comment (tyxla) [inline client/state/jetpack/connection/test/reducer.js]: I'd reword `status request` to `disconnect request` here.

File: client/state/jetpack/connection/test/reducer.js [hunk 4]
```diff
@@ -146,4 +150,72 @@ describe( 'reducer', () => {
...
+					siteId
+				};
+			const stateOut = disconnectRequestsReducer( deepFreeze( stateIn ), action );
+			expect( stateOut ).to.eql( {
+				...REQUESTS_FIXTURE,
+				[ siteId ]: true
+			} );
+		} );
+
+		it( 'should set requesting to false for the specified site when status request completes successfully', () => {
```
Comment (tyxla) [inline client/state/jetpack/connection/test/reducer.js]: I'd reword `status request` to `disconnect request` here.

File: client/state/jetpack/connection/test/reducer.js [hunk 5]
```diff
@@ -146,4 +150,72 @@ describe( 'reducer', () => {
...
+					siteId
+				};
+			const stateOut = disconnectRequestsReducer( deepFreeze( stateIn ), action );
+			expect( stateOut ).to.eql( {
+				...REQUESTS_FIXTURE,
+				[ siteId ]: false
+			} );
+		} );
+
+		it( 'should set requesting to false for the specified site when status request completes unsuccessfully', () => {
```
Comment (tyxla) [inline client/state/jetpack/connection/test/reducer.js]: I'd reword `status request` to `disconnect request` here.

**Comment by enejb** [comment]
Thanks for the review @ebinnion and @tyxla 
Just rebased this will merge after the tests pass.
